src/organisms/PermissionPage.jsx

```diff
@@ -41,18 +44,35 @@ const BtnInvertedContainer = styled.div`
   margin-top: 45px;
 `;
 
-const PermissionPage = ({ getLocation, handleLocation }) => {
+export const UPDATE_PERMISSIONS = gql`
+  mutation setRejectedPermissions($rejectedPermissions: Boolean!) {
+    setRejectedPermissions(rejectedPermission: $rejectedPermission) @client {
+      Permission {
+        rejectedPermission
+      }
+    }
+  }
+`;
+
+const PermissionPage = () => {
   const history = useHistory();
+  const [gpsMutation] = location.gpsMutationHook();
+  const [setPermissions] = useMutation(UPDATE_PERMISSIONS);
 
   const handleAccept = () => {
-    localStorage.setItem("firstTime", true);
-    getLocation(handleLocation, history);
+    location.setGpsCache(gpsMutation);
+    setPermissions({ variables: { rejectedPermission: false } });
+    //set current location (mutation)
```

Comment:
just be sure to remove the commented out code as it gets closer to the end of Labs.